horace_core/sqw/@SQWDnDBase/private/smooth_dnd.m

```diff
@@ -0,0 +1,157 @@
+function dout = smooth_dnd(din, xunit, varargin)
+% Smooths a 1,2,3 or 4 dimensional dataset
+%
+%Syntax:
+%   >> dout = smooth_dnd (din, xunit, width, shape)
+%
+% Input:
+% ------
+%   din     Input dataset structure (data_sqw_dnd)
+%   xunit   If true, unit of measure is length of a unit along each axis
+%           If false, unit of measure is pixels
+%   width   Vector that sets the extent of the smoothing along each dimension.
+%          The interpretation of width depends on the argument 'shape' described
+%          below.
+%           If width is scalar, then the value is applied to all dimensions
+%
+%           e.g. if din is a 3-dimensional dataset, valid arguments for width might be:
+%                width = [2,4,5]    % 2, 3, 5 along the 1st, 2nd and 3rd dimensions
+%                width = 4.5        % 4.5 applied to all dimensions
+%           Invalid choices for 3-dimensions are
+%                width = [2,3]      % invalid number of dimensions
+%
+%   shape   Shape of smoothing function
+%               'hat'           Hat function
+%                                   - width gives FWHH along each dimension
+%                                   - width = 1,3,5,...;  n=0 or 1 => no smoothing
```

Comment:
this description is pretty opaque up to the semicolon (what does it do any better than line 18?) and then does n= mean the width value or that no smoothing is applied for those dimensionalities?

Reply:
The functions mapped by the shape argument here are defined in `SQWDnDBase/private/smooth_func_xxx`
I have no more knowledge of their behaviour other than this comment and the header in those functions.